template-for-repository/proofs/Makefile.common

```diff
@@ -138,6 +138,22 @@ ifneq ($(strip $(EXTERNAL_SAT_SOLVER)),)
 endif
 CHECKFLAGS += $(USE_EXTERNAL_SAT_SOLVER)
 
+# Job pools
+# For version of Litani that are new enough (where `litani print-capabilities`
+# prints "pools"), proofs for which `EXPENSIVE = true` is set can be added to a
+# "job pool" that forces only a single "expensive" proof to run at a time. All
+# other proofs will be built in parallel as usual. To enable this feature, set
+# the ENABLE_POOLS variable when running Make, like
+#     `make ENABLE_POOLS=true report`
+
+ifeq ($(strip $(ENABLE_POOLS)),)
+  POOL =
+else ifeq ($(strip $(EXPENSIVE)),)
+  POOL =
+else
+  POOL = --pool expensive
+endif
+
```

Comment:
>  If either enable_pools or expensive is not set, then the value of the other is thrown away and pool is set to empty, but if both have any value whatsoever the value is thrown away and pool is set to request an expensive pool

ENABLE_POOLS is a global setting, passed in by the run script. EXPENSIVE is set per proof.

If ENABLE_PROOFS is unset, then EXPENSIVE is ignored, since we're saying we don't want to use the feature even for proofs that have been explicitly marked as EXPENSIVE.

The user-facing variable is EXPENSIVE, thus I have added a commit to document its use: see "Document the use of EXPENSIVE in template Makefile". The run script handles ENABLE_PROOFS. So I've added a comment in Makefile.common about how it's used (see "Improve documentation for job pools feature"), but I've made it clear that the run script deals with it.

> Is EXPENSIVE set in an individual proof Makefile or can it be set in a top-level Makefile.defines?

You could set it top-level, but why would you? If you want to limit the concurrency of *all* proofs, you would use `-j`, there is no sense in marking all proofs as being EXPENSIVE.

Thanks!